spi_nand_flash/include/spi_nand_flash.h

```diff
@@ -47,7 +52,7 @@ esp_err_t spi_nand_flash_init_device(spi_nand_flash_config_t *config, spi_nand_f
  * @param sector_id The id of the sector to read.
  * @return ESP_OK on success, or a flash error code if the read failed.
  */
-esp_err_t spi_nand_flash_read_sector(spi_nand_flash_device_t *handle, uint8_t *buffer, dhara_sector_t sector_id);
+esp_err_t spi_nand_flash_read_sector(spi_nand_flash_device_t *handle, uint8_t *buffer, sector_t sector_id);
```

Comment:
One thing that is a bit confusing is that this header includes:
- functions which operate on top of the wear levelling layer (dhara) — like `spi_nand_flash_read_sector`
- functions which operate directly with the NAND storage — `spi_nand_flash_is_bad_block`.

I think it would be better to separate these two. You kind of already do this by having a header file spi_nand_api.h for these lower level functions (which is, a bit confusingly, not part of the public API, despite the file name).

Reply:
For debugging and diagnostics, I added the spi_nand_flash_is_bad_block() function to get bad block stats. I didn’t want to expose spi_nand_api directly but still needed to access its information, so I created this public API to serve that purpose.

Or perhaps I could create a separate source file for the diagnostic APIs. This file would use spi_nand_api (which I’ll rename), and these APIs could then be used in debug applications.

Reply:
Well, you could expose the "lower-level" API as well, just in a different header file and with a different prefix. This could help in writing test cases. This would also resolve the naming issue around spi_nand_api.h.

Reply:
Right now I have removed this API from public header. I will add new header while adding diagnostic application in separate PR.

Reply:
Added nand_impl_wrap.h header which will expose lower-level APIs. PTAL.